website/docs/r/secrets_sync_association.html.md

```diff
@@ -0,0 +1,79 @@
+---
+layout: "vault"
+page_title: "Vault: vault_secrets_sync_association resource"
+sidebar_current: "docs-vault-resource-secrets-sync-association"
+description: |-
+  Triggers a sync operation in Vault and links a secret to an existing destination
+---
+
+# vault\_secrets\_sync\_association
+
+Triggers a sync operation in Vault and links a secret to an existing destination.
+Requires Vault 1.16+. *Available only for Vault Enterprise*.
+
+~> **Important** All data provided in the resource configuration will be
+written in cleartext to state and plan files generated by Terraform, and
+will appear in the console output when Terraform runs. Protect these
+artifacts accordingly. See
+[the main provider documentation](../index.html)
+for more details.
+
+For more information on associations, please refer to the Vault
+[documentation](https://developer.hashicorp.com/vault/docs/sync#associations).
+
+## Example Usage
+
+```hcl
+resource "vault_mount" "kvv2" {
+  path        = "kvv2"
+  type        = "kv"
+  options     = { version = "2" }
+  description = "KV Version 2 secret engine mount"
+}
+
+resource "vault_kv_secret_v2" "token" {
+  mount = vault_mount.kvv2.path
+  name  = "token"
+  data_json = jsonencode(
+    {
+      dev  = "B!gS3cr3t",
+      prod = "S3cureP4$$"
+    }
+  )
+}
+
+resource "vault_secrets_sync_gh_destination" "gh" {
+  name = "gh-dest"
+}
```

Comment:
Should the example use explicit token, repo & owner fields?

Reply:
Updated!